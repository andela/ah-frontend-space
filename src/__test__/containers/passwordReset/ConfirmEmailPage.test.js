import React from 'react';
import { shallow } from 'enzyme';

import {
  ConfirmEmailPage,
  mapStateToProps,
  mapDispatchToProps,
} from '../../../containers/passwordReset/ConfirmEmailPage';

function renderConfirmEmailPage(args) {
  const defaultProps = {
    loading: false,
    onChange: jest.fn(),
    onSubmit: jest.fn(),
    actions: {},
  };
  const props = { ...defaultProps, ...args };
  return shallow(<ConfirmEmailPage {...props} />);
}

describe('ConfirmEmailPage', () => {
  it('should match snapshot', () => {
    const wrapper = renderConfirmEmailPage();
    wrapper.setState({
      user: {
        email: 'test@email.com',
      },
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('should map state to props', () => {
    const state = {
      user: {
        email: 'email@email.com',
      },
    };
    expect(mapStateToProps(state)).toEqual({
      user: {
        email: 'email@email.com',
      },
      loading: false,
    });
  });

  it('should map dispatch to the confirmEmail action', () => {
    const user = {
      email: 'email@email.com',
    };
    const dispatch = jest.fn();

    mapDispatchToProps(dispatch).actions.confirmEmailSuccess(user);
    expect(dispatch.mock.calls[0][0]).toEqual({ type: 'CONFIRM_EMAIL_SUCCESS' });
  });

  it('should handle submit successfully', () => {
    const wrapper = renderConfirmEmailPage();
    wrapper.setState({
      user: {
        email: 'test@email.com',
      },
    });
    wrapper.setProps({
      actions: { confirmEmail: jest.fn() },
    });
    const wrapperInstance = wrapper.instance();
    const event = { preventDefault: jest.fn() };
    jest.spyOn(wrapperInstance, 'onSubmitHandler');
    wrapperInstance.onSubmitHandler(event);

    expect(wrapperInstance.onSubmitHandler).toBeCalled();
  });

  it('should handle onChange event', () => {
    const wrapper = renderConfirmEmailPage();
    wrapper.setState({
      target: {
        name: 'email',
        value: 'test@email.com',
      },
    });
    const wrapperInstance = wrapper.instance();
    const event = {
      target: {
        name: 'email',
        value: 'test@email.com',
      },
    };
    jest.spyOn(wrapperInstance, 'onChangeHandler');
    wrapperInstance.onChangeHandler(event);

    expect(wrapperInstance.onChangeHandler).toBeCalled();
  });
});
