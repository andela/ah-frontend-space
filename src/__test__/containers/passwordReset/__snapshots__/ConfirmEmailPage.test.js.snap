// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmEmailPage should match snapshot 1`] = `ShallowWrapper {}`;
